tests/test_company_list_and_persist_contracts.py

```
@@ -0,0 +1,52 @@
+    await store.create_company(Company(name="Acme", domain="acme.com", owner_id="u1"))
+    result = await store.list_companies(owner_id="u1", limit=10)
+    assert isinstance(result, list)
+    assert len(result) == 1 and result[0].name == "Acme"
+
+
+def test_company_is_frozen_and_activity_goes_through_model_copy():
+    company = Company(name="Acme", domain="acme.com", owner_id="u1")
+
+    # Frozen: direct attribute assignment must fail (this is why the old
+    # `company.activity_log = []` persist path silently did nothing).
+    with pytest.raises(Exception):
```

Ruff (0.15.15)

[warning] Do not assert blind exception: `Exception`

(B017)